plugin/pkg/shoot/validator/admission.go

```diff
@@ -569,6 +575,55 @@ func (c *validationContext) validateProvider() field.ErrorList {
 	return allErrs
 }
 
+func (c *validationContext) validateAPIVersionForRawExtensions() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Provider.InfrastructureConfig); usesInternalVersion {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "infrastructureConfig"), gvk, "must not use apiVersion 'internal'"))
+	}
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Provider.ControlPlaneConfig); usesInternalVersion {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "controlPlaneConfig"), gvk, "must not use apiVersion 'internal'"))
+	}
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Networking.ProviderConfig); usesInternalVersion {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "networking", "providerConfig"), gvk, "must not use apiVersion 'internal'"))
+	}
+
+	for i, worker := range c.shoot.Spec.Provider.Workers {
+		if usesInternalVersion, gvk := checkRawExtension(worker.ProviderConfig); usesInternalVersion {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "provider", "workers").Index(i).Child("providerConfig"), gvk, "must not use apiVersion 'internal'"))
```

Comment:
This path is used in several places, could as well be extracted into a variable.
```suggestion
		workerPath := field.NewPath("spec", "provider", "workers").Index(i)
		if usesInternalVersion, gvk := checkRawExtension(worker.ProviderConfig); usesInternalVersion {
			allErrs = append(allErrs, field.Invalid(workerPath.Child("providerConfig"), gvk, "must not use apiVersion 'internal'"))
```

Reply:
done.